responses/configuration/set.go

```diff
@@ -60,33 +54,26 @@ func (sc *SetConfigResponse) UnmarshalJSON(data []byte) error {
 	}
 
 	//If verified, copy the fields
-	if test.Code != nil {
-		sc.Code = *test.Code
+	if test.Success != false {
+		sc.Success = test.Success
 	}
 
 	if test.Description != nil {
 		sc.Description = *test.Description
 	}
 
-	if test.ExpectedValues != nil {
-		sc.ExpectedValues = test.ExpectedValues
-	}
-
 	sc.isValidated, err = sc.Validate()
 
 	return err
 }
 
 // Validate satisfies the Validator interface
 func (sc SetConfigResponse) Validate() (bool, error) {
-	if sc.Code == "" {
-		return false, models.NewErrContractInvalid(fmt.Sprintf("invalid Code %s", sc.Code))
+	if sc.Success != true {
```

Comment:
Is this correct?  Isn't a valid Success state false?

Reply:
That is exactly what I had in mind when I had encoded the logical test initially, as follows...
`if sc.Success != true || sc.Success != false {`
...and the feedback I got from the IDE was that `expression can be simplified` (This appears to be a case where IDE feedback should be ignored).

Reply:
Validate()'s purpose, I think, is to determine whether or not the structure is in a valid state. 

Your code is looking at the SetConfigResponse.Success value -- which could either be true or false -- and only considering the structure to be valid when the Success is true.  I think the structure is valid when Success is false as well.